### ISSUE TITLE
NodeFailures on SLURM w/ version 2.6

### ISSUE THREAD

**Issue by medcelerate** [aws/aws-parallelcluster#1709]
**Environment:**
 - 2.6.0
```
[aws]
aws_region_name = us-east-1

[global]
cluster_template = pcprod
update_check = true
sanity_check = true

[aliases]
ssh = ssh {CFN_USER}@{MASTER_IP} {ARGS}

[cluster pcprod]

key_name = s4-sinaia-valinor-nphi
base_os = alinux2
cluster_type = spot
scheduler = slurm

master_instance_type = m4.4xlarge
compute_instance_type = c5.12xlarge
compute_root_volume_size = 500
master_root_volume_size = 800

initial_queue_size = 1
max_queue_size = 800
maintain_initial_size = true

s3_read_resource = arn:aws:s3:::s4-sinaia-valinor-nphi/*
s3_read_write_resource = arn:aws:s3:::s4-sinaia-valinor-nphi/*

tags = {"RunSchedule": "24x7", "PC": "PCPROD"}

extra_json = { "cluster" : { "ganglia_enabled" : "yes" } }

fsx_settings = performancefs
efs_settings = generalfs
scaling_settings = sc
cw_log_settings = pcprodlog

post_install = https://gitlab.com/iidsgt/parallel-cluster/-/raw/c3d78aef623eeeb642f938bad378901a4df293ca/ProdPostInstall.py

post_install_args = "{cromwell_db_user} {cromwell_db_password} {S3_Key} {S3_Secret}"

vpc_settings = pcluster

[cw_log pcprodlog]
enable = true
retention_days = 14

[efs generalfs]
shared_dir = /efs
performance_mode = generalPurpose

[fsx performancefs]
shared_dir = /fsx
storage_capacity = 9600
#imported_file_chunk_size = 1024
deployment_type = SCRATCH_2
#export_path = s3://parallelcluster/fsx
#import_path = s3://parallelcluster
weekly_maintenance_start_time = 5:00:00


[scaling sc]
scaledown_idletime = 30

[vpc pcluster]
### Excluded values
```

**Bug description and how to reproduce:**
With ondemand or spot instances nodes are consistently being put into a failed state, or being marked as down and the job being marked as complete when it is not complete.
**Additional context:**

Cloudwatch nodewatcher lines for a failed node.
```
15:25:04
2020-03-23 15:25:04,651 INFO [slurm:is_node_down] Node is in state: ''

15:25:04
2020-03-23 15:25:04,652 WARNING [nodewatcher:_poll_wait_for_node_ready] Node reported as down

15:25:14
2020-03-23 15:25:14,682 INFO [slurm:is_node_down] Node is in state: ''

15:25:14
2020-03-23 15:25:14,682 WARNING [nodewatcher:_poll_wait_for_node_ready] Node reported as down

15:25:24
2020-03-23 15:25:24,710 INFO [slurm:is_node_down] Node is in state: ''

15:25:24
2020-03-23 15:25:24,710 WARNING [nodewatcher:_poll_wait_for_node_ready] Node reported as down

15:25:34
2020-03-23 15:25:34,740 INFO [slurm:is_node_down] Node is in state: ''

15:25:34
2020-03-23 15:25:34,740 WARNING [nodewatcher:_poll_wait_for_node_ready] Node reported as down

15:25:34
2020-03-23 15:25:34,740 ERROR [nodewatcher:_terminate_if_down] Node is marked as down by scheduler or not attached correctly. Terminating...
```
Node Logs
[i-0ffa05db478debd93.tar.gz](https://github.com/aws/aws-parallelcluster/files/4370697/i-0ffa05db478debd93.tar.gz)

**Comment by rexcsn** [comment]
Hi @medcelerate ,

Thank you for opening this issue, I see that you have posted [comment](https://github.com/aws/aws-parallelcluster/issues/1327#issuecomment-602690335) in another post as well.

We have recently seen an issue with the master's network being overwhelmed when there are large number of compute instances, which causes all sorts of problems in `sqswatcher` and `nodwatcher`. We will be looking into this issue, it would help if you can also provide the logs on the master node, especially `jobwatcher` and `sqswatcher` logs.

In the meantime it might be worth for you to try using a larger compute instance type in order to decrease the number of instances while maintaining the number of cores.

**Comment by medcelerate** [comment]
Thanks for the reply! I should mention this is happening even on just 1 or
two nodes. No mpi just single jobs on each node. I will add the logs.

**Comment by medcelerate** [comment]
[sqswatcher.txt](https://github.com/aws/aws-parallelcluster/files/4388189/sqswatcher.txt)
[jobwatcher.txt](https://github.com/aws/aws-parallelcluster/files/4388191/jobwatcher.txt)

**Comment by medcelerate** [comment]
Here are the files you requested from the master node.

**Comment by medcelerate** [comment]
@rexcsn Have the logs been of any usefulness?

**Comment by rexcsn** [comment]
Hi @medcelerate ,

Unfortunately, I did not have much time to look.

Here is what I have found so far:
* From the `jobwatcher` logs I am seeing nodes being placed in `down*` and `unk*` from time to time. Not sure of the reason.
* Node successfully added to scheduler config:
```
2020-03-18 23:58:57,598 INFO [sqswatcher:_parse_sqs_messages] Processing parallelcluster:COMPUTE_READY event for instance Host(instance_id='i-0ffa05db478debd93', hostname='ip-10-0-24-231', slots='48', gpus=0)
2020-03-18 23:58:57,599 INFO [slurm:_write_node_list_to_file] Writing updated /opt/slurm/etc/slurm_parallelcluster_nodes.conf with the following nodes: ['NodeName=dummy-compute[1-793] CPUs=48 State=FUTURE\n', 'NodeName=ip-10-0-28-209 CPUs=48 State=UNKNOWN\n', 'NodeName=ip-10-0-23-177 CPUs=48 State=UNKNOWN\n', 'NodeName=ip-10-0-16-181 CPUs=48 State=UNKNOWN\n', 'NodeName=ip-10-0-17-138 CPUs=48 State=UNKNOWN\n', 'NodeName=ip-10-0-17-19 CPUs=48 State=UNKNOWN\n', 'NodeName=ip-10-0-27-226 CPUs=48 State=UNKNOWN\n', 'NodeName=ip-10-0-24-231 CPUs=48 State=UNKNOWN\n']
```
* For the particular instance `i-0ffa05db478debd93`, I don't see any obvious error/exception in `nodewatcher` log, but `nodewatcher` is unable to retrieve the node's state from scheduler: 
```
2020-03-18 23:59:06,860 INFO [slurm:is_node_down] Node is in state: 'allocated'
...
2020-03-19 00:00:07,236 INFO [slurm:is_node_down] Node is in state: ''
```
* Since `nodewatcher` is unable to retrieve node state, it shuts the node down after 5 minutes, which is expected: `2020-03-19 00:05:08,179 ERROR [nodewatcher:_terminate_if_down] Node is marked as down by scheduler or not attached correctly. Terminating...`
* The node is correctly removed from slurm configuration: `2020-03-19 00:06:59,277 INFO [sqswatcher:_process_sqs_messages] Successfully processed event REMOVE for instance Host(instance_id='i-0ffa05db478debd93', hostname='ip-10-0-24-231', slots=None, gpus=None)`

I would say that overall `pcluster` workflow looks ok, and the main question is why `nodewatcher` was unable to retrieve the node's state from scheduler, especially when it is able to see the node in `allocated` at first.
Slurm logs might help in this case, please attach `/var/log/slurmctld.log` from master and `/var/log/slurmd.log` from compute if possible.

On the other hand, are you doing any action to modify/update nodes in slurm either manually or from your post-install script? Please confirm that the same problem will persist if you remove the post-install script and use on-demand.

Thank you!

**Comment by medcelerate** [comment]
@rexcsn I can confirm the issue does still occur with ondemand instances and spot instances. The only thing we are doing is installing docker. Attached is the slurmctld log. Slurmd logs are hard to get from a node once it is marked as NODE Failure. It appears most of the issues come from when a node is marked as failure. 

[slurmctld.log](https://drive.google.com/open?id=1jJCtqQ151Ryoo_g2u3PunzmH_F5QMrIC)

**Comment by medcelerate** [comment]
Attached is the slurmd from the last available node.
[slurmd.log](https://github.com/aws/aws-parallelcluster/files/4422450/slurmd.log)

**Comment by medcelerate** [comment]
A quick update on this, submitted a few jobs that simply require using the aws CLI, and noticed that when network load becomes high the node is unable to be accessed. SSH to the nodes is incredibly slow and the nodes are eventually marked as nodefailure and killed. I'm not sure what the best way to resolve the networking issue would be between the nodes. We are using the c5.4xlarge which should have up to 10gb bandwidth. This happens if writing to EFS or FSX storage.

**Comment by woodshop** [comment]
I've been watching this thread because I recently set up parallelcluster and experienced compute node issues. My compute nodes were consistently being terminated, always about 10 minutes after starting. This behavior occurred with more than one scheduler. Changing 'scaledown_idletime' had no effect.

I too was installing docker in my startup scripts. My issue may be unrelated to this one, but considering we're both installing docker, I thought I'd share.  I found that when I moved my docker installation from post_install to pre_install the compute nodes became stable. 

I haven't had time to investigate any more thoroughly. I'm wondering if parallelcluster and docker try to use any of the same ports.

**Comment by medcelerate** [comment]
This is interesting, let me try moving docker to pre-install instead. This seems to occur when using docker and when not using docker, but I will give it a go.

**Comment by medcelerate** [comment]
@woodshop I went ahead and moved the docker install to pre-install script and it did give the nodes some extra life time, but still having the same issue, nodes are dying around 15 minutes.

**Comment by woodshop** [comment]
Ah too bad @medcelerate, I guess it was worth a try though.

**Comment by medcelerate** [comment]
@woodshop One thing to mention I updated the config for all the compute nodes but not the master node, did you see this change when you updated the master as well?

**Comment by woodshop** [comment]
@medcelerate I did a complete teardown with `pcluster delete`. I don't think I tested the scenario where different configs were used for Master and Compute. In case it helps, I've pasted below my anonymized config, pre_, and post_install scripts. (My post_install script is mainly used to import our home directory and other pre-existing efs volumes into the new "cluster".)

config
```bash
[global]
cluster_template = default
update_check = true
sanity_check = true

[aws]
aws_region_name = us-east-1

[cluster default]
base_os = alinux2
compute_instance_type = p2.xlarge
initial_queue_size = 0
maintain_initial_size = true
master_instance_type = m4.large
max_queue_size = 12
placement_group = DYNAMIC
pre_install = XXX
post_install = XXX
s3_read_resource = arn:aws:s3:::XXX/*
scheduler = slurm
vpc_settings = default
key_name = XXX
shared_dir = /shared
scaling_settings = default
ebs_settings = default

[vpc default]
vpc_id = XXX
vpc_security_group_id = XXX
additional_sg = XXX
master_subnet_id = XXX
use_public_ips = false

[scaling default]
scaledown_idletime = 10

[ebs default]
volume_size = 64
```

parallel-cluster-pre-install.sh
```bash
#!/bin/bash
set -x
yum update -y

# install docker with gpu support
amazon-linux-extras install -y docker
distribution=$(. /etc/os-release;echo $ID$VERSION_ID)
curl -s -L https://nvidia.github.io/nvidia-docker/$distribution/nvidia-docker.repo | tee /etc/yum.repos.d/nvidia-docker.repo
yum install -y nvidia-container-toolkit
systemctl restart docker
```

parallel-cluster-post-install.sh
```bash
#!/bin/bash
set -x
. "/etc/parallelcluster/cfnconfig"

yum update -y
yum install -y amazon-efs-utils

file_system_id_1=XXX
file_system_id_3=XXX
file_system_id_4=XXX
efs_mount_point_3=/data/audio
efs_mount_point_4=/home

mkdir -p "${efs_mount_point_3}"
users=( XXX )
for user in "${users[@]}"
do
    mkdir -p "${efs_mount_point_4}/${user}"
done


echo "${file_system_id_3}:/ ${efs_mount_point_3} efs tls,_netdev" >> /etc/fstab
for user in "${users[@]}"
do
    echo "${file_system_id_4}:/${user} ${efs_mount_point_4}/${user} efs tls,_netdev" >> /etc/fstab
done

mount -a -t efs,nfs4 defaults

echo "%wheel ALL=(ALL) NOPASSWD:ALL" > /etc/sudoers.d/wheel
groupadd -g 1001 XXX
groupadd -g 1002 XXX
groupadd -g 1003 XXX
groupadd -g 1004 XXX
groupadd -g 1005 XXX
groupadd docker

useradd -M -u 1001 -g XXX -G izo,docker,wheel -b "${efs_mount_point_4}" XXX
useradd -M -u 1002 -g XXX -G izo,docker -b "${efs_mount_point_4}" XXX
useradd -M -u 1003 -g XXX -G izo,docker -b "${efs_mount_point_4}" XXX
useradd -M -u 1004 -g XXX -G izo,docker -b "${efs_mount_point_4}" XXX

# install extra packages
yum install -y tmux htop

# if Master node, mount file_system_id_1 and copy to /shared
case "${cfn_node_type}" in
    MasterServer)
        mkdir /mnt/efs
        mount -t efs -o tls ${file_system_id_1}:/ /mnt/efs
        cp -r /mnt/efs/* /shared
        umount /mnt/efs
        ;;
    ComputeFleet)
        ;;
    *)
        ;;
esac
```

**Comment by medcelerate** [comment]
@woodshop thanks! I'm starting up a second parallel cluster with the same config and see if we see the same issues with the same config on master and compute.

**Comment by medcelerate** [comment]
@woodshop @rexcsn So I created a new cluster following woodshops guidance however at exactly 10 minutes the node which the tasks are running on is marked as down and removed.

**Comment by medcelerate** [comment]
More information, I spun up another cluster with dynamic placement group, had the same issue. Furthermore when the node went down it did not create any logs in /home/logs. @rexcsn Could this be an issue with the controller node and compute nodes being in different availability zones?

**Comment by medcelerate** [comment]
@woodshop Did you ever use a public/private subnet configuration or just a single subnet?

**Comment by woodshop** [comment]
Hey @medcelerate, with regard to multiple availability zones, th PC wiki suggest that there could be problems with EFS:
https://github.com/aws/aws-parallelcluster/wiki/EFS-issues

The docs also advise a single availability zone is better for performance, via a placement group:
https://github.com/aws/aws-parallelcluster/wiki/Best-Practices

As for my subnet configuration, I'm using a single private subnet for both master and compute. 
The subnet allows outbound internet connections via an AWS-managed NAT Gateway. The subnet is reachable within our company's VPN via a site-to-site VPN connection. (I did not personally set up NAT, VPN, and subnets; these were arranged by our IT.)

A few other things... is your parallelcluster version up-to-date? v6.1 was just released a few days ago. Have you made sure that your DHCP option set is configured properly (https://docs.aws.amazon.com/parallelcluster/latest/ug/networking.html)? Also, make sure to run `pcluster create` with the `-nr` flag. This way the CloudFormation stack won't be immediately torn down if creation fails, and will preserve any logs in CloudWatch.

**Comment by medcelerate** [comment]
@woodshop Definitely not related to EFS, this issue happened prior. The automated vpc creation in parallel cluster put the public and private subnets in separate zones for some reason. The cluster creation doesn't fail but when the nodes die they don't leave much trace. 

There is no logs.gz but I do have the relevant ones saved in cloudwatch. Both dns options are set in the vpc.

I am running version 2.6.1 now and am seeing the same issues. It seems like this issue may be networking related but I am unsure what other methods of debugging could be used to trace where the issue is arrising.

**Comment by woodshop** [comment]
@medcelerate I guess your problem is probably not related to availability zones, but you should be able to force them to be in the same zone by setting 
```bash
[cluster default]
...
placement = cluster
placement_group = DYNAMIC
...
```

The next thing I would inspect is the security group assignment. Is it possible that the sg is too restrictive? I am using two security groups. The primary group allows all incoming traffic from itself. This should guarantee that the compute and master nodes can talk to each other. My second security group is configured to allow incoming connections from subnets associated with the VPN. You could probably combine these two groups.

Finally, even though your compute nodes are terminating, the logs should persist for 2 weeks in CloudWatch (as long as you don't delete the cluster). Look for any hints in "/aws/parallelcluster/<cluster name>/ip-<compute instance ip separated with hyphens>.<compute instance id>.cfn-init".

**Comment by medcelerate** [comment]
@woodshop setting `placement=cluster` actually causes an error in setup. I have to try running it again, but believe it is due to the public and private subnets being in a different AZ. I checked both the security groups and they allow all traffic between them, both inbound and outbound. I do have the logs in cloudwatch, but there is nothing of interest. The only thing is nodewatcher almost immediately receives a signal that the node is marked down by slurm. I can share that here.

**Comment by woodshop** [comment]
@medcelerate Ah, I didn't understand that your subnets are in different AZs. I can't find anything in the docs that say this isn't supported, but all of the network configurations in the following page show the subnets residing in the same AZ.

https://docs.aws.amazon.com/parallelcluster/latest/ug/networking.html

Is it possible to retry using the same AZ for both subnets?

**Comment by medcelerate** [comment]
@woodshop Let me see if I can create a VPC with both in the same AZ, the interesting thing is that this VPC was create with the pcluster configure command so it put the two subnets in different availability zones.

**Comment by medcelerate** [comment]
@woodshop Created a new vpc and waiting to see if jobs die, but in the meantine, interesting information. When a job is running there is what is happening in node watcher.

```
2020-04-14 02:11:48,639 INFO [slurm_commands:get_pending_jobs_info] Retrieved 0 pending jobs
2020-04-14 02:11:48,721 INFO [utils:get_asg_settings] ASG min/desired/max: 0/1/800
2020-04-14 02:11:48,722 INFO [nodewatcher:_poll_instance_status] Instance had no job for the past 40 minute(s)
2020-04-14 02:12:48,632 INFO [nodewatcher:_refresh_cluster_properties] Refreshing cluster properties
2020-04-14 02:12:48,739 INFO [utils:get_asg_settings] ASG min/desired/max: 0/1/800
2020-04-14 02:12:48,753 INFO [slurm:is_node_down] Node is in state: 'idle'
2020-04-14 02:12:48,764 INFO [slurm:has_jobs] Found the following running jobs:

2020-04-14 02:12:48,771 INFO [slurm_commands:get_pending_jobs_info] Retrieved 0 pending jobs
2020-04-14 02:12:48,891 INFO [utils:get_asg_settings] ASG min/desired/max: 0/1/800
2020-04-14 02:12:48,893 INFO [nodewatcher:_poll_instance_status] Instance had no job for the past 41 minute(s)
2020-04-14 02:13:48,663 INFO [slurm:is_node_down] Node is in state: 'idle'
2020-04-14 02:13:48,674 INFO [slurm:has_jobs] Found the following running jobs:

2020-04-14 02:13:48,680 INFO [slurm_commands:get_pending_jobs_info] Retrieved 0 pending jobs
2020-04-14 02:13:48,800 INFO [utils:get_asg_settings] ASG min/desired/max: 0/1/800
2020-04-14 02:13:48,801 INFO [nodewatcher:_poll_instance_status] Instance had no job for the past 42 minute(s)
2020-04-14 02:14:48,731 INFO [slurm:is_node_down] Node is in state: ''
2020-04-14 02:14:48,731 WARNING [nodewatcher:_poll_wait_for_node_ready] Node reported as down
```

**Comment by demartinofra** [comment]
Hi @medcelerate 

sorry for our late reply. let me try to help drilling down the issue you are facing.

First things I would do are the following:
- Create the same cluster without any pre/post install scripts and see if the issue occurs also in that case.
- Pick a master node with enhanced networking such as m5n.4xlarge
- Always use `placement = compute` especially if you require different instance types for the master and compute fleet.
- Unfortunately picking a specific AZ is a current limitation of `pcluster configure` we are aware of. I think for now you could stick to the single subnet setup so that we do not change to many variables at once.

**Comment by medcelerate** [comment]
So far I have tried

Changing the master node to c5n.4xlarge - Same Issue

Used placement = compute

Manually added another private subnet in the same AZ as the public subnet.

I will try creating without pre and post install scripts, though our
workload are heavily reliant on docker.


@demartinofra

**Comment by demartinofra** [comment]
> I will try creating without pre and post install scripts, though our workload are heavily reliant on docker.

sure I don't want you to get rid of those but I would like to understand if the problem comes from the custom script so that we can focus on debugging that one and understand how it conflicts with ParallelCluster.

**Comment by medcelerate** [comment]
@demartinofra Ok, is there any recommendation for tools to simulate heavy file system usage. I have tested before and high cpu, high memory jobs work fine, only when there start to be writes to fsx or efs that issues start arising.

**Comment by demartinofra** [comment]
You could try to run something like this:
```
#!/bin/bash

count=1
while true; do 
  ((count++))
  if !((count%10000)); then
    echo "$(date)"
    dd if=/dev/zero of=/path/to/dir/filei1M_$$.$count.txt count=1 bs=1048576
    sleep 0.01
  fi
done
```

And use `sbatch` to submit a script that looks something like this (where of course you need to set nodes and tasks)
```
cat job.sh
#!/bin/bash
#BATCH --cpus-per-task=1
set -xe

srun /shared_area/cpu_io_intensive_job.sh
```

My guess is that EFS and FSx are not the real culprits here but rather networking and shared master EBS drive are. We noticed in the past that ganglia performs some aggressive I/O as the number of nodes in the cluster grows while Slurm requires significant networking as the cluster scales. Also the stdout and stderr of the jobs are by default redirected to the job working directory and if you are using the user home this is shared over NFS.

Something that might help is building a CloudWatch dashboard and look at some of the relevant metrics. Here is a quick example: https://gist.githubusercontent.com/demartinofra/4a1fa0ff84726114f1037746151ba2c6/raw/36aff4ea75d7e09c2ed0ed114c519c10e87c2c52/ParallelCluster%2520CW%2520dashboard%2520example

**Comment by medcelerate** [comment]
@demartinofra I went ahead and the script on one node with two separate jobs and I am not seeing the usual error messages in nodewatcher, normally it would be giving blank node states and then getting a response that the node is down from the controller. So now this makes me curious what in my pre and post install scripts could be causing the issue.

**Comment by medcelerate** [comment]
This is what nodewatcher gives currently
```
2020-04-16 21:05:58,448 INFO [slurm:has_jobs] Found the following running jobs:
                 4   compute   run.sh ec2-user  R       6:05      1 ip-10-0-34-78
                 5   compute   run.sh ec2-user  R       2:45      1 ip-10-0-34-78
2020-04-16 21:05:58,448 INFO [nodewatcher:_poll_instance_status] Instance has active jobs.
2020-04-16 21:06:58,388 INFO [slurm:is_node_down] Node is in state: 'mixed'
2020-04-16 21:06:58,398 INFO [slurm:has_jobs] Found the following running jobs:
                 4   compute   run.sh ec2-user  R       7:05      1 ip-10-0-34-78
                 5   compute   run.sh ec2-user  R       3:45      1 ip-10-0-34-78
2020-04-16 21:06:58,398 INFO [nodewatcher:_poll_instance_status] Instance has active jobs.
2020-04-16 21:07:58,447 INFO [slurm:is_node_down] Node is in state: 'mixed'
2020-04-16 21:07:58,457 INFO [slurm:has_jobs] Found the following running jobs:
                 4   compute   run.sh ec2-user  R       8:05      1 ip-10-0-34-78
                 5   compute   run.sh ec2-user  R       4:45      1 ip-10-0-34-78
2020-04-16 21:07:58,457 INFO [nodewatcher:_poll_instance_status] Instance has active jobs.
2020-04-16 21:08:58,454 INFO [slurm:is_node_down] Node is in state: 'mixed'
2020-04-16 21:08:58,465 INFO [slurm:has_jobs] Found the following running jobs:
                 4   compute   run.sh ec2-user  R       9:05      1 ip-10-0-34-78
                 5   compute   run.sh ec2-user  R       5:45      1 ip-10-0-34-78
2020-04-16 21:08:58,465 INFO [nodewatcher:_poll_instance_status] Instance has active jobs.
2020-04-16 21:09:58,513 INFO [slurm:is_node_down] Node is in state: 'mixed'
2020-04-16 21:09:58,524 INFO [slurm:has_jobs] Found the following running jobs:
                 4   compute   run.sh ec2-user  R      10:05      1 ip-10-0-34-78
                 5   compute   run.sh ec2-user  R       6:45      1 ip-10-0-34-78
2020-04-16 21:09:58,524 INFO [nodewatcher:_poll_instance_status] Instance has active jobs.
```

**Comment by medcelerate** [comment]
The main things that are installed are:
Docker
GLauth (LDAP: Port 389)
Cromwell (Workflow Manager: Port 8088)
Nslcd (LDAP Client: Port 389)

We previously saw this issue in a much more limited capacity when using docker and may have been fixed in 2.6.1, so I'm curious which one of these services may be the source of the issue. GLauth most likely makes the most network calls as each user is authenticated.

**Comment by rexcsn** [comment]
Hi @medcelerate,

Were you able to figure out the issue or still seeing the issue with `pcluster` v2.6.1?

Recently we saw some customers hitting issues because something in their setup was causing `slurmctld` to respond slowly. We check node status simply by executing `sinfo`, and if you are seeing slow response when executing `sinfo` when using your specific setup, that might cause the command to retrieve node status to timeout and fail.
Curiously, it seems from your log that the command used to retrieve node status did not fail, but the node status is empty, which does not correspond to this case exactly, but might be worth checking anyway.

Thank you!

**Comment by medcelerate** [comment]
Yes we are still having the issue, something is causing a big lag in the network, if I remove the ldap and docker components it seems to solve the issue, but I don't comprehend why ldap would be causing an issue.

**Comment by lukeseawalker** [comment]
Hi @medcelerate,
were you able to figure out which component installed in the pre or post install scripts is causing the problem?

**Comment by medcelerate** [comment]
So it seems to have to do with the LDAP server. I have yet to find a solution.

**Comment by demartinofra** [comment]
I'm going to resolve this since there hasn't been any progress lately. Starting from ParallelCluster 2.9 we have heavily rearchitected our integration with Slurm scheduler. It might be worth giving it a try and feel free to open a new issue in case you are still blocked by this performance issue.
